[AndreChia] iP

AndreChia opened the pull request:
# **Phone 📱**  
> _“Your mind is for having ideas, not holding them.”_ – **David Allen**

**Phone** is your intelligent task management assistant, designed to help you **track tasks efficiently**. It's:

- ✅ **Text-based**  
- ✅ **Easy to learn**  
- ✅ **Super FAST 🚀**  

## **How to Use It?**  
1. **Download** the JAR file from [here](https://github.com/AndreChia/ip/releases).  
2. **Run** it using the command:  
`java -jar phone.jar`
3. Start managing tasks effortlessly! 🎉

**Features:**
- 📌 Managing ToDos, Deadlines, and Events
- 🔍 Finding tasks quickly using keywords
- 📅 Smart date & time parsing

> Why use PhoneBot?
> It helps you focus on your work instead of remembering what needs to be done. 💡

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

AndreChia committed:
Added Phone.java

AndreChia committed:
Added Echo

AndreChia committed:
Add, List

AndreChia committed:
Mark as Done

Commit:
half done through L4

Commit:
half done through L4

AndreChia committed:
ToDos, Events, Deadlines

AndreChia committed:
Updated User Testing

AndreChia committed:
Handle Errors

AndreChia committed:
Delete Added

AndreChia committed:
Added Exception Handling and Error Checking

AndreChia committed:
Merge branch 'master' of https://github.com/AndreChia/ip

AndreChia committed:
Add Save

AndreChia committed:
Merge branch 'branch-Level-7'

AndreChia committed:
Add Dates and Times

AndreChia committed:
Merge branch 'branch-Level-8'

AndreChia committed:
Add OOP

AndreChia committed:
Merge branch 'A-MoreOOP'

AndreChia committed:
Add Packages

AndreChia committed:
Merge branch 'A-Packages'

AndreChia committed:
Merge remote-tracking branch 'upstream/add-gradle-support'

AndreChia committed:
Add JUnit

AndreChia committed:
Merge branch 'A-JUnit'

AndreChia committed:
Add Jar File

AndreChia committed:
Merge branch 'A-Jar'

AndreChia committed:
Format into Checkstyle

AndreChia committed:
Merge branch 'A-Checkstyle'
Added checkstyle

aquaimpact commented on src/main/java/phone/Ui.java:
```diff
@@ -0,0 +1,66 @@
+     * @return User command.
+     */
+    public String readCommand() {
+        return scanner.nextLine().trim();
```
Wow! I didn't think of implementing the `readCommand()` this way!

aquaimpact commented on src/main/java/phone/Parser.java:
Keep up the good work with the naming convention across all files! They were clear and succinct!

aquaimpact commented on src/main/java/phone/command/AddCommand.java:
```diff
@@ -0,0 +1,56 @@
+    public void execute(TaskList tasks, Ui ui, Storage storage) {
+        Task task;
+        try {
+            switch (type) {
```
This statement could be moved into your parser file instead of being abstracted as a separate AddCommand. Placing it in the parser would streamline command handling and improve code organization.

aquaimpact commented on src/main/java/phone/command/InvalidCommand.java:
This file can be refactored as an exception class (... extends Exception) instead of an InvalidCommand class. Additionally, you might consider splitting it into multiple custom exception classes, such as FileHandlingException, to improve error handling and make debugging more efficient.

aquaimpact reviewed (commented):
Overall, LGTM! 👍 Great job with maintaining clear and consistent naming conventions. It looks like you're ahead in terms of Checkstyle and testing as well—keep up the excellent work!

AndreChia committed:
WIP on master: b3f4678 Merge branch 'A-Checkstyle' Added checkstyle

AndreChia committed:
index on master: b3f4678 Merge branch 'A-Checkstyle' Added checkstyle

AndreChia committed:
Add GUI

AndreChia committed:
Merge branch 'recovered-branch'
Reverting changes

cscms03 commented on src/main/java/phone/DialogBox.java:
Great JavaDoc comments.

cscms03 commented on src/main/java/phone/Launcher.java:
```diff
@@ -0,0 +1,12 @@
+/**
+ * A launcher class to work around JavaFX classpath issues.
+ */
+public class Launcher {
```
Did not think of creating separate Launcher class for main method. Great idea

cscms03 commented on src/main/java/phone/Parser.java:
```diff
@@ -0,0 +1,48 @@
+        case "delete":
+            return new DeleteCommand(arguments);
+        default:
+            return new InvalidCommand();
```
I think ENUM use is very good in terms of readability. Maybe could add toLowerCase() if not done?

cscms03 reviewed (commented):
Overall, the code follows coding standard (such as import order, naming conventions). Very neat code!

AndreChia committed:
Resolve merge conflicts and fixed GUI implementation with TaskList

AndreChia committed:
Fixed merge conflicts

AndreChia committed:
Removed return statement that caused a bug

AndreChia committed:
Add assertions

AndreChia committed:
Merge pull request #1 from AndreChia/branch-A-Assertions

Add assertions to AddCommand class

AndreChia committed:
Modify AddCommand for Code Qualtiy Improvements

AndreChia committed:
Resolve merge conflict in AddCommand.java after pulling

AndreChia committed:
Implement client saving functionality

Added saveClients method to Storage to persist client data to a text file. Modified AddClientCommand to call saveClients after adding a client to ensure data persistence.

AndreChia committed:
Use Java Streams to format task list in getFormattedTaskList

The current implementation of getFormattedTaskList uses a for-loop to iterate through the task list and append each task to a StringBuilder.

AndreChia committed:
Finalize PhoneGPT for submission

AndreChia committed:
Improve PhoneGPT User Guide
